nixos/doc/manual/development/option-types.section.md

```diff
@@ -337,6 +337,16 @@ Composed types are types that take a type as parameter. `listOf
     value of type *`to`*. Can be used to preserve backwards compatibility
     of an option if its type was changed.
 
+`types.recursive` *`f`*
+
+:   A recursively defined type based on the fixed point of `f`.
+    As an example, you could define the type of boolean [rose trees](https://en.wikipedia.org/wiki/Rose_tree)
+    as `with types; recursive (t: either bool (listOf t))`. If you used
+    `lib.fix` instead of `recursive`, you would run into infinite recursions
+    as the module system tries to determine the description and suboptions of this type.
+
+    This is useful for defining recursive formats such as JSON.
+
```

Comment:
![screenshot-2022-12-29-134511](https://user-images.githubusercontent.com/4323933/209952963-f859f522-33c1-4d4a-a611-af3643a4f503.png)

Reply:
Boolean trees are also a subset of JSON :P this is not meant to be super applicable anyway, it's just a basic example.

Not sure why we'd need `types.json`

Reply:
I do like @roberth's proposal with `attrsOf` better. I also think we can have a paragraph just explaining what that then means. Like how `"x"`, `{ a = "x"; }`, `{ a.b = "x"; c = "x"; }` are then all accepted.

@roberth Yeah I also don't see why we'd want `types.json`, I haven't seen the need for `.type` without also `.generator`